moped-editor/src/views/projects/projectView/ProjectComponentEdit.js

```diff
@@ -221,7 +235,7 @@ const ProjectComponentEdit = ({
                 component_subtype: component?.component_subtype ?? null,
               },
             },
-            subcomponents: componentSubcomponents,
+            subcomponents: [...currentSubcomponents, ...componentSubcomponents],
```

Comment:
@sergiogcx i uncovered a tricky bug here that we somehow hadn't noticed before. Subcomponent changes were not being saved properly for components which had a required subtype, because the subtype's subcomponents were being overwritten here. I have some ideas for a more elegant fix, but this gets it done :/

Reply:
@johnclary Ah, thanks for finding this bug. It makes sense, and your fix is elegant as-is imo. Happy to hear the other ideas you have.

Reply:
in that case I'll leave it as-is ;) glad i was able to get familiar with this reducer—it's awesome!